test/app/gh-6203-fiber-join-with-timeout.test.lua

```diff
@@ -0,0 +1,21 @@
+env = require('test_run')
+test_run = env.new()
+fiber = require('fiber')
+
+timeout = 10
+function sleep(timeout) \
+    channel:get() \
+    fiber.sleep(timeout) \
+end
+
+channel = fiber.channel()
+f = fiber.create(sleep, timeout)
+f:set_joinable(true)
+channel:put(true)
+f:join(0.1) -- Join failed because of timeout
+
+timeout = 0.2
+f = fiber.create(sleep, timeout)
+f:set_joinable(true)
+channel:put(true)
+f:join("xxx") --error: 'usage: fiber:join([timeout])'
```

Comment:
There's also check of negative numeric values. I guess it is worth testing here.